indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -339,6 +339,17 @@ public static Configuration injectSystemProperties(Configuration conf)
     return conf;
   }
 
+  public static Configuration injectSystemAndJobProperties(Configuration conf, HadoopDruidIndexerConfig config)
+  {
+    Configuration updatedConf = injectSystemProperties(conf);
+    HadoopIngestionSpec schema = config.getSchema();
+
+    for (final Map.Entry<String, String> entry : schema.getTuningConfig().getJobProperties().entrySet()) {
+      updatedConf.set(entry.getKey(), entry.getValue());
+    }
```

Comment:
there are two overloaded `JobHelper.injectSystemProperties(Job)` and `JobHelper.injectSystemProperties(Configuration)` and first one uses the second. Similarly `HadoopDruidIndexerConfig.addJobProperties(Job)` can be refactored and add a `HadoopDruidIndexerConfig.addJobProperties(Configuration)`. Then you can use that. makes sense ?